### PR TITLE
Test exploration

### PR DESCRIPTION
**Pull request by michaelcfanning**

### CODE REVIEW

File: src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs [hunk 1]
```diff
@@ -30,6 +30,8 @@ public abstract class MultithreadedAnalyzeCommandBase<TContext, TOptions> : Plug
         internal ConsoleLogger _consoleLogger;
         internal ConcurrentDictionary<string, IAnalysisLogger> _analysisLoggerCache;
 
+        internal static int? s_numberOfThreadsForTesting;
```
Comment (michaelcfanning) [inline src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:33]: >s_numberOfThreadsForTesting [](http://example.com/codeflow?start=29&length=27)

You should pick up this value from the context.

File: src/Test.UnitTests.Sarif.Driver/TestRule.cs [hunk 2]
```diff
@@ -245,11 +256,15 @@ public static void RaiseExceptionViaReflection()
...
         {
-            return new IOption[] { Behaviors, UnusedOption };
+            return new IOption[] { Behaviors, UnusedOption, ErrorsCount };
         }
 
         private const string AnalyzerName = TestRuleId + "." + nameof(TestRule);
 
+        public static PerLanguageOption<uint> ErrorsCount { get; } =
+            new PerLanguageOption<uint>(
+                AnalyzerName, nameof(ErrorsCount), defaultValue: () => { return 15; });
```
Comment (michaelcfanning) [inline src/Test.UnitTests.Sarif.Driver/TestRule.cs:266]: >15 [](http://example.com/codeflow?start=80&length=2)

Restore this to 1, that was the previous behavior. Or provide a justification for the updated value.

File: src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs [hunk 3]
```diff
@@ -676,6 +676,54 @@ public void AnalyzeCommandBase_EndToEndAnalysisWithPostUri()
...
+            mockFileSystem.Setup(x => x.DirectoryGetFiles(It.IsAny<string>(), specifier)).Returns(files);
+            mockFileSystem.Setup(x => x.FileExists(It.Is<string>(s => s.EndsWith(specifier)))).Returns(true);
+            mockFileSystem.Setup(x => x.DirectoryEnumerateFiles(It.IsAny<string>(),
+                                                                It.IsAny<string>(),
+                                                                It.IsAny<SearchOption>())).Returns(files);
+            mockFileSystem.Setup(x => x.FileOpenRead(It.IsAny<string>())).Returns(mockStream.Object);
+
+            for (int i = 0; i < 100; i++)
+            {
+                var options = new TestAnalyzeOptions
```
Comment (michaelcfanning) [inline src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs:707]: >TestAnalyzeOptions [](http://example.com/codeflow?start=34&length=18)

Add an option here for configuration, 'Options.xml'. Update the Mock object to return a string representation that has your option. Find a clean way to create a config file with just this option.

File: src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs [hunk 4]
```diff
@@ -592,7 +596,7 @@ internal AggregatingLogger InitializeLogger(AnalyzeOptionsBase analyzeOptions)
 
             if (filePath != null)
             {
-                context.TargetUri = new Uri(filePath);
+                context.TargetUri = new Uri(filePath, UriKind.RelativeOrAbsolute);
```
Comment (michaelcfanning) [inline src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:599]: >UriKind.RelativeOrAbsolute [](http://example.com/codeflow?start=54&length=26)

File an issue, add a test.

File: src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs (hunk 1 above)
Comment (eddynaka) [reply src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:33]: removed.

**Comment by eddynaka** [comment]
Closing this since we merged this pr in the following:
https://github.com/microsoft/sarif-sdk/pull/2447

**Pull request closed without merging**